docs/DeveloperGuide.md

```diff
@@ -2,37 +2,615 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the 
+original source as well}
 
```

Comment:
I think it would be better to include a table of contents and links to the headers for easier navigation